builtin/providers/aws/resource_aws_lambda_function_test.go

```diff
@@ -157,6 +254,61 @@ func testAccCheckAwsLambdaFunctionArnHasSuffix(function *lambda.GetFunctionOutpu
 	}
 }
 
+func testAccCheckAwsLambdaSourceCodeHash(function *lambda.GetFunctionOutput, expectedHash string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		c := function.Configuration
+		if *c.CodeSha256 != expectedHash {
+			return fmt.Errorf("Expected code hash %s, got %s", expectedHash, *c.CodeSha256)
+		}
+
+		return nil
+	}
+}
+
+func testAccCreateZipFromFiles(files map[string]string, zipFile *os.File) error {
+	zipFile.Truncate(0)
+	zipFile.Seek(0, 0)
+
+	w := zip.NewWriter(zipFile)
+
+	for source, destination := range files {
+		f, err := w.Create(destination)
+		if err != nil {
+			return err
+		}
+
+		fileContent, err := ioutil.ReadFile(source)
+		if err != nil {
+			return err
+		}
+
+		_, err = f.Write(fileContent)
+		if err != nil {
+			return err
+		}
+	}
+
+	err := w.Close()
+	if err != nil {
+		return err
+	}
+
+	return w.Flush()
+}
+
+func createTempFile(prefix string) (string, *os.File, error) {
+	f, err := ioutil.TempFile(os.TempDir(), prefix)
+	if err != nil {
+		return "", nil, err
+	}
+
+	pathToFile, err := filepath.Abs(f.Name())
+	if err != nil {
+		return "", nil, err
+	}
+	return pathToFile, f, nil
+}
```

Comment:
It's the caller's responsibility to remove the result of `ioutil.TempFile`. Should defer an `os.Remove` when we use this.